geo_activity_playground/webui/static/square-planner.js

```diff
@@ -0,0 +1,65 @@
+document.addEventListener("DOMContentLoaded", function () {
```

Comment:
I'm not very experienced with JavaScript. What is the advantage of this compared to just having the `<script>` tag very late in the DOM?

Reply:
I've extracted the inline script into a separate file. The rationale is a simple separation of concerns[^1].

The script is loaded with the `defer` attribute. That means, the script will be loaded non-blocking and is executed before the `DOMContentLoaded` event is fired by the browser.

The event handler ensures the DOM is parsed and referenced elements are available before the code runs.

[^1]: A further optimization is to load GeoJSON asynchronously after the initial render. This would improve page load times significantly for large number of tiles (I've around 80k level 17 tiles which result in long load times of around five seconds for the Square Planner).

Reply:
I think for the square planner, it would also make sense to use server rendered tiles at some point.